manifests/resource/map.pp

```diff
@@ -67,6 +76,7 @@
   Variant[Array, Hash] $mappings,
   Optional[String] $default         = undef,
   Enum['absent', 'present'] $ensure = 'present',
+  Array[String] $includes           = [],
```

Comment:
Will this work if it's an empty array, since you're not checking if it's empty in the template? Will this create an empty `includes` line?
Maybe this should be `Optional[Array[String]] $includes`?

Reply:
Because there's a default value you can omit it ; it acts as an optional parameter. 
So if you pass an empty Array it will be the same as passing nothing and use the default value: it won't iterate nothing into the loop in template so nothing will be written (no empty includes line).
I agree with you since it is finally an optional parameter that declaring it explicitly as Optional would be better (but in that case the test in the template will became mandatory).